forem/forem
# Publicly indicate if someone edits their comments

Signed-off-by: Efereyan Karen Simisola <thedevkaren@gmail.com>



## What type of PR is this? (check all applicable)


## Description
This PR aims to add a public indication of edited comments. This prevents gaslighting. It's helpful to stop situations where User A says something, then User B responds, User A modifies their comment so that User B’s response doesn’t make sense. Ideally, if someone edits their comments, we should publicly indicate it in the comment with the word "(edited)" appearing near the date of the comment.

## Related Tickets & Documents

Closes #10373

## QA Instructions, Screenshots, Recordings

Follow these steps to test the changes(These commands should be run via the terminal)
- `psql PracticalDeveloper_development`
- `select id, LEFT(processed_html, 30) as processed_html, edited_at from comments order by id;` (This returns a list of comments ordered by id. The id of the specific comment to be edited should be noted as it will go into the parenthesis in the next command)
- `update comments set edited_at = current_timestamp where id in (id of comment to be edited);` (This should return the keyword UPDATED along with the number of comments that were updated
- `select id, LEFT(processed_html, 30) as processed_html, edited_at from comments order by id;`
- `exit`

Once the preview is loaded again, the edited span should be added to the comment that was just edited

The following images illustrate the changes
![1](https://user-images.githubusercontent.com/56835876/94879996-b4fcd100-0459-11eb-8b39-c95bb1cde379.png)
![2](https://user-images.githubusercontent.com/56835876/94880092-f2f9f500-0459-11eb-83f9-8106ccbecafb.png)

![3](https://user-images.githubusercontent.com/56835876/94880109-ff7e4d80-0459-11eb-8ee0-68b120b9545c.png)

![4](https://user-images.githubusercontent.com/56835876/94880119-07d68880-045a-11eb-8531-40904d03fcfc.png)
![8th bit brooklyn](https://user-images.githubusercontent.com/56835876/94880194-305e8280-045a-11eb-91dc-89fccaebc127.png)
![8th bit edited](https://user-images.githubusercontent.com/56835876/94880236-44a27f80-045a-11eb-894a-84a8ff81b4b1.png)

## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/5GoVLqeAOo6PK/giphy.gif)
